library_test.py

```diff
@@ -54,22 +55,21 @@ async def main() -> None:
     else:
         api = VodafoneStationSercommApi(args.router, args.username, args.password)
 
-    exc = True
     try:
-        await api.login()
-        exc = False
-    except ModelNotSupported:
-        print("Model is not supported yet for router", api.host)
-    except CannotConnect:
-        print("Cannot connect to router", api.host)
-    except AlreadyLogged:
-        print("Only one user at a time can connect to router", api.host)
-    except GenericLoginError:
-        print("Unable to login to router", api.host)
-    finally:
-        if exc:
-            await api.close()
-            exit(1)
+        try:
+            await api.login()
+        except ModelNotSupported:
+            print("Model is not supported yet for router", api.host)
+        except CannotConnect:
+            print("Cannot connect to router", api.host)
+        except AlreadyLogged:
+            print("Only one user at a time can connect to router", api.host)
+        except GenericLoginError:
+            print("Unable to login to router", api.host)
+    except VodafoneError:
```

Comment:
Looks generally good, i think this should be a generic Exception when we add to HA. We will re-raise it, but we want to close on all exceptions.